### DISCUSSION TITLE
New @times directive

### DISCUSSION THREAD

**Discussion by marknuyens** [laravel/framework#50623] [Ideas]
Hello everyone,

I've encountered a few occasions where I wanted to repeat some part of my Blade markup. For example:

```blade
@for ($i = 0; $i < 2; $i++)
    <x-icon class="size-10 fill-red-500 />
@endfor
```

I figured that this could be easily converted into a `@times` directive:

```php
Blade::directive('times', fn (int $amount) => "<?php for (\$i = 0; \$i < {$amount}; \$i++): ?>");
Blade::directive('endtimes', fn () => '<?php endfor; ?>');
```

This would result in:

```blade
@times(2)
    <x-icon class="size-10 fill-red-500 />
@endtimes
```

Arguments against may include simplicity, but that may very well also be an argument _for_ using this. Also, if you're anything like me, remembering the `for` loop syntax can sometimes be tricky - this would help simplify this.

Anyway, curious to what y'all think, :smile: 

Thanks!

**Comment by shaedrich** [comment]
This could be simplified to:
```php
@foreach (range(0, 1) as $i)
    <x-icon class="size-10 fill-red-500 />
@endforeach
```

And I wonder, how many cases of production code need the exact same HTML (from inside `@times`) and aren't in fact more complicated again.

**Comment by marknuyens** [comment]
You're right by suggesting this feature might not be the most popular, but I did run into a few situations and I was curious if anyone else had a similar experience. Other than that, I do think my directive would provide a more elegant syntax as opposed to `range()`, IMHO.

But again, I would totally understand if there was little need for this. Just throwing it out there ;-)
